### ISSUE TITLE
scheme.xml wrongly generate default value for CURRENT_TIMESTAMP(6)

### ISSUE THREAD

**Issue by Brekeke** [propelorm/Propel2#1905]
If I set in SQL default value for timestamp data type column as `current_timestamp(6)` it's generated like:

```<column name="created" phpName="Created" type="TIMESTAMP" size="6" defaultValue="current_timestamp(6)"/>```

But when is called module:build from that scheme.xml it return this error:

![conhost_2022-08-26_12-47-21](https://user-images.githubusercontent.com/11869992/186887818-c8721b85-0a0a-4c3d-a5e9-48074e835a81.png)

When I change defaultValue to defaultExpr in scheme.xml manually it works. Working part of code looks like:

```<column name="created" phpName="Created" type="TIMESTAMP" size="6" defaultExpr="current_timestamp(6)"/>```

So issue is in generator of scheme.xml which wrongly use defaultValue for current_timestamp(6) and another similar values instead of defaultExpr like it do for normal current_timestamp without length definition.

If is possible fix it, should be nice, thanks :-)

**Comment by mringler** [comment]
Yes, that is a bit confusing. But I think it is the expected behavior according to the [docs](http://propelorm.org/documentation/reference/schema.html#column-attributes):
>- `defaultValue` The default value that the object will have for this column in the PHP instance after creating a “new Object”. This value is always interpreted as a string.
>- `defaultExpr` The default value for this column as expressed in SQL. This value is used solely for the “sql” target which builds your database from the schema.xml file. The defaultExpr is the SQL expression used as the “default” for the column.

I think `defaultValue` has to be something that PHP understands, like `new DateTime()`, and if you want to use a default value on database level, you need to use `defaultExpr`, as you do now, and it seems to work.

So it sounds like it is working as expected, but maybe I just misunderstand your issue, then feel free to clarify.

**Comment by adjenks** [comment]
`now()` as a default value in postgres in a date column also fails when trying to run `model:build`.

**Comment by adjenks** [comment]
It looks like most of my date columns set to `now()` were generated using `defaultExpr` but just one of them wasn't. This particular one has brackets around it, so perhaps it was interpreted as being a value for some reason. It's defined like so:
`date_valid date NOT NULL DEFAULT (now())::date`
